enhancements/agent-installer/development-environment.md

```diff
@@ -0,0 +1,305 @@
+[dev-scripts](https://github.com/openshift-metal3/dev-scripts) repository and refactor 
+components as needed for the new project. In addition we will add base case e2e test flows
+integrated into Openshift CI to get the project started.
+
+## Motivation
+
+The new agent based installer project will need a standard development environment for
+the team to use. We will need an e2e testing framework for validation in Openshift CI. 
+In this enhancement we will explain our desire to reuse existing frameworks for the new 
+project while limiting the scope of modifications to them to the new project's requirements.
+
+In this space we have two existing frameworks that could be used for this new project. These 
+frameworks are:
+* [dev-scripts](https://github.com/openshift-metal3/dev-scripts): Created by the Metal Platform 
+(IPI) team to set up environments to install OCP both for local development and in Openshift CI.
+* [assisted-test-infra](https://github.com/openshift/assisted-test-infra): Created by the Assisted
+Installer team to run full e2e test flows.
+
+Looking at existing e2e test flow steps from these two frameworks it was determined that 
+in dev-scripts test flows the beginning steps are very similar to the development
+flows we desire for the agent installer. In contrast the assisted-test-infra test flows
+set up extra things in the development environment that is undesirable for the agent installer.
+
+### Goals
+
+* Using this development environment in the new ephemeral agent based installer image project.
+* Re-factoring parts of dev-scripts that are needed for this ephemeral agent based installer.
+* Convergence of the underlying scripts and environment setup code between Metal Platform (IPI) 
+team and the new Agent Installer teams.
+
+### Non-Goals
+
+* Re-factoring dev-scripts and assisted-test-infra for reasons not related to the new 
+ephemeral agent based installer image project.
+* Overall convergence between all the installer development teams of scripts and e2e test flows.
+
+## Proposal
+
+Create an development environment for the ephemeral agent based installer image in the 
+[dev-scripts](https://github.com/openshift-metal3/dev-scripts) repository and refactor 
+components as needed for the new project.
+
+### User Stories
+
+* As a developer, I need to set up an environment so that I can simulate the installation of a 
+users first cluster using the ephemeral based agent installer image.
+
+* As a developer, I need to launch the built ephemeral agent based installer image so that I can test 
+and validate this image during development.
+
+* As a developer, I need to install Openshift using the ephemeral agent based installer image 
+to validate the agent installer.
+
+* As a developer, I need to check cluster installation progress of the Openshift cluster being
+installed by agent installer and when this installation is complete to validate the Openshift
+cluster.
```

Comment:
It's not really included in how you write a user story but I plan to use the existing Openshift tests to validate the installed cluster.

Reply:
I think that (in the long term) the number of configurations * OpenShift versions * e2e conformance suits required for an exhaustive vaidation will make it impractical to verify it locally, leaving this responsibility to the CI rehearsals.
During the development activity I'd like to have a quick / gross way to verify the current changes, so probably we could identify a number of minimum must-have checks required - and possibly common to all the possible configurations (ie cluster is up), but that's probably something we could leave as an implementation details and here just leave "verify that installation is complete"

Reply:
I like this idea of defined minimum checks. That would help with onboarding new developers and learning to develop against OCP. This could be something we define for the future... We should at least put it in the backlog.

My desire to have the configurations local is more about running the configuration I'm currently working on while I'm developing. I want to have that ability to do that on demand and easily. For instance if I'm working on a feature that is specific to SNO clusters, I would only run SNO based flows locally while developing. When I push up my work the CI will run compact cluster and HA clusters checks against my code to make sure I didn't break anything. After that is finished I might work on a feature that is focused on a different type of cluster configuration than SNO so I wouldn't run SNO based flows. I realize this is my personal preference in developing but I think we can accommodate both styles without much issue.

Reply:
I also just realized this thread is more about validations of the OCP clusters than configurations.

Local validations I'd like to see:
- Kubeconfig is created and accessible
- Control Plane came up
- Cluster is installed
- A generic way to capture the error output when cluster fails to install